### DISCUSSION TITLE
flux dry-run failed: Istio VirtualService doesn't support dryRun

### DISCUSSION THREAD

**Discussion by olivercp3** [fluxcd/flux2#3652] [Q&A] [answered]
error info:
```shell
NAME                            REVISION                SUSPENDED       READY   MESSAGE

kustomization/flux-system       master@sha1:b0085663    False           False   VirtualService/default/jd-adminservice dry-run failed, reason: ServiceUnavailable, error: the server is currently unable to handle the request (patch virtualservices.networking.istio.io jd-adminservice)
```

I can apply my backend service by `kubectl apply`, and my backend service are working
so why will this happend? i am sure my deploy file is works，It does not depend on other resources

**Comment by stefanprodan** [comment, marked as answer]
To simulate what Flux does do `kubectl apply --server-side --dry-run=server -f ./some.yaml`, given that error, the Istio admission webook is down.

**Reply by olivercp3** [reply]
error: networking.istio.io/v1alpha3, Kind=VirtualService doesn't support dryRun

**Reply by olivercp3** [reply]
thank you
